### ISSUE TITLE
Compilation

### ISSUE THREAD

**Issue by ir-g** [tj/luna#63]
Would it be possible to add the ability to compile Luna code to machine code. (possibly by embedding code in the VM)

**Comment by autohawk** [comment]
is this project died?

**Comment by aisk** [comment]
+ 1

**Comment by glhrmfrts** [comment]
I'd like to consider that as an option, luna has everything to be a compiled language. However I think the prior target is building a pure interpreted language.

**Comment by isaacrg** [comment]
@glhrmfrts Indeed. I merely suggested it because of the ease of distribution a compiled application has over bundling a VM.

**Comment by tj** [comment]
that was the original plan, but compiled would be fun too! I'd like a crossover between Go and Rust personally, something inbetween would be a nice fit, but that's also a ridiculous amount of work haha

**Comment by isaacrg** [comment]
@tj Yeah - It would be a lot of effort. The syntactical appearance of Luna really appeals to me as something to write small compiled web apps in, for easy local distribution - but the sheer amount of stuff that would need to be implemented makes that impossible.

**Comment by glhrmfrts** [comment]
Well, the good thing is that we are still in the parsing phase, so there's plenty of time to discuss this more deeply.
